sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/models/DiskEncryptionSet.java

```diff
@@ -0,0 +1,236 @@
+        interface WithGroup extends GroupableResource.DefinitionStages.WithGroup<WithEncryptionType> { }
+
+        /**
+         * The stage of a disk encryption set definition allowing to set the disk encryption set type.
+         */
+        interface WithEncryptionType {
+            /**
+             * Set the disk encryption set type.
+             * @param type the type of key used to encrypt the data of the disk
+             * @return the next stage of the definition
+             */
+            WithKeyVault withEncryptionType(DiskEncryptionSetType type);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to associate with an Azure key vault.
+         */
+        interface WithKeyVault {
+            /**
+             * Associates with the disk encryption set an Azure key vault by its resource ID.
+             * @param keyVaultId resource ID of the Azure key vault
+             * @return the next stage of the definition
+             */
+            WithKeyVaultKey withExistingKeyVault(String keyVaultId);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to associate with an Azure key vault key.
+         */
+        interface WithKeyVaultKey {
+            /**
+             * Associate with the disk encryption set an Azure key vault key by its ID.
+             * @param keyId ID of the Azure key vault key
+             * @return the next stage of the definition
+             */
+            WithSystemAssignedManagedServiceIdentity withExistingKey(String keyId);
+        }
+
+        /**
+         * The stage of a disk encryption set definition allowing to enable System Assigned (Local) Managed Service
+         * Identity.
+         */
+        interface WithSystemAssignedManagedServiceIdentity {
+            /**
+             * Specifies that System Assigned (Local) Managed Service Identity needs to be enabled in the disk
+             * encryption set.
+             *
+             * @return the next stage of the definition
+             */
+            WithSystemAssignedIdentityBasedAccessOrCreate withSystemAssignedManagedServiceIdentity();
+        }
+
+        /**
+         * The stage of the System Assigned (Local) Managed Service Identity enabled disk encryption set allowing to set
+         * access role for the key vault.
+         */
+        interface WithSystemAssignedIdentityBasedAccessOrCreate extends WithCreate {
+            /**
+             * Specifies that disk encryption set's system assigned (local) identity should have the given access
+             * (described by the role) on the current Azure key vault that's associated with it.
+             * @param builtInRole access role to assigned to the disk encryption set's local identity
+             * @return the next stage of the definition
+             */
+            WithCreate withSystemAssignedIdentityBasedAccessToCurrentKeyVault(BuiltInRole builtInRole);
```

Comment:
Vault has 2 configure for auth, one is accessPolicy, another is RBAC.

This likely only work for latter. And for this case, should be provide a overload with default role (I am not sure the exact role, maybe `KEY_VAULT_CRYPTO_SERVICE_ENCRYPTION_USER`).

Reply:
Thanks, I'll check the access policy!

Reply:
Adding access policy to key vaults involves keyvault code. Since currently compute does not depend on keyvault, do we want to add that dependency?
For cli, they do:
> az keyvault set-policy -n $keyVaultName \
-g $rgName \
--object-id $desIdentity \
--key-permissions wrapkey unwrapkey get

Reply:
Emm, maybe not yet.

For now, probably just write in the RBAC method, that if user uses accessPolicy, we expect them to call Vault for accessPolicy.

Reply:
OK.